public/app/features/templating/template_srv.ts

```diff
@@ -275,30 +295,23 @@ export class TemplateSrv implements BaseTemplateSrv {
   }
 
   replaceWithText(target: string, scopedVars?: ScopedVars) {
+    deprecationWarning('template_srv.ts', 'replaceWithText()', 'replace(), and specify the :text format');
+
     if (!target) {
-      return target;
+      return target ?? '';
     }
 
-    let variable;
     this.regex.lastIndex = 0;
 
-    return target.replace(this.regex, (match: any, var1: any, var2: any, fmt2: any, var3: any) => {
-      if (scopedVars) {
-        const option = scopedVars[var1 || var2 || var3];
-        if (option) {
-          return option.text;
-        }
-      }
-
-      variable = this.getVariableAtIndex(var1 || var2 || var3);
-      if (!variable) {
+    const withoutFormat = target.replace(this.regex, (match, var1, var2, fmt2, var3, fieldPath, fmt3) => {
+      const format = fmt2 ?? fmt3;
+      if (!format) {
         return match;
       }
-
-      const value = this.grafanaVariables[variable.current.value];
-
-      return typeof value === 'string' ? value : variable.current.text;
+      return match.replace(`:${format}`, '');
     });
```

Comment:
Why is this still needed? if a user has specified a format then that should be used always. So just returning `this.replace(target, scopedVars, 'text');` should be enough

Reply:
Or should we change that logic?

Reply:
yea, think that one is not correct (or can be changed). 

The use case for replaceWithText is really only in panel title, and legend / display name inputs. And the reason for all the other formats are to work with query syntaxes. So if someone want's glob formatting then that's what they get.